object/slicer/options.go

```diff
@@ -0,0 +1,77 @@
+package slicer
+
+import (
+	"io"
+
+	"github.com/nspcc-dev/neofs-sdk-go/session"
+)
+
+// Options groups all options of SliceDataIntoObjects.
+type Options struct {
+	payload io.Reader
+
+	objectPayloadLimit uint64
+
+	currentNeoFSEpoch uint64
+
+	sessionToken *session.Object
+
+	attributes []string
+}
+
+// SetPayload allows to provide source of the data to be associated with the
+// object. If set, SliceDataIntoObjects will read object payload from the r
+// until io.EOF.
+//
+// See also SetObjectPayloadLimit.
+func (x *Options) SetPayload(r io.Reader) {
+	x.payload = r
+}
+
+// SetObjectPayloadLimit specifies data size limit for produced physically
+// stored objects.
+//
+// NeoFS stores the corresponding value in the network configuration. Ignore
+// this option if you don't (want to) have access to it.
+//
+// By default, single object is limited by 32KB. SliceDataIntoObjects uses this
+// value to enforce the maximum object payload size limit described in the NeoFS
+// Specification. If the total amount of data exceeds the specified limit,
+// SliceDataIntoObjects applies the slicing algorithm described within the same
+// specification. The outcome will be a group of "small" objects containing a
+// chunk of data, as well as an auxiliary linking object. All derived objects
+// are written to the parameterized ObjectWriter. If the amount of data is
+// within the limit, one object is produced.
+//
+// SetObjectPayloadLimit SHOULD NOT be used if payload source is not specified
+// using SetPayload (i.e. no payload).
+//
+// Note that SliceDataIntoObjects can write multiple objects, but only returns
+// the root object ID.
+func (x *Options) SetObjectPayloadLimit(l uint64) {
+	x.objectPayloadLimit = l
+}
+
+// SetCurrentNeoFSEpoch allows to specify NeoFS epoch to be written to
+// the resulting object's metadata as a creation epoch.
+func (x *Options) SetCurrentNeoFSEpoch(e uint64) {
+	x.currentNeoFSEpoch = e
```

Comment:
OK to have an override, but again we can take it from the current data by default.

Reply:
Current slicer is abstracted from where objects are going to be written. Thus, it is far from certain that the caller has a connection to the NeoFS network. But if it has, it can fetch these values and statically set them. Solution depends on https://github.com/nspcc-dev/neofs-sdk-go/pull/382#discussion_r1118938647.